### PR TITLE
*: use full sampling as default analyze

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Problem Summary:

This pr change the default analyze tach to full sampling based one.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

Change the default analyze version. And fixing the unit-test changes.

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

File: statistics/histogram.go [hunk 1]
```diff
@@ -1289,6 +1285,7 @@ func (idx *Index) expBackoffEstimation(sc *stmtctx.StatementContext, coll *HistC
 	//   2. Sort them and choose the first 4 most selective filter and the corresponding selectivity is sel_1, sel_2, sel_3, sel_4 where i < j => sel_i < sel_j.
 	//   3. The final selectivity would be sel_1 * sel_2^{1/2} * sel_3^{1/4} * sel_4^{1/8}.
 	// This calculation reduced the independence assumption and can work well better than it.
+	logutil.BgLogger().Warn("exp backoff", zap.String("range", fmt.Sprintf("%v", indexRange)), zap.String("col ids", fmt.Sprintf("%v", colsIDs)))
```
Comment (qw4990) [inline statistics/histogram.go:1288]: Should we print column names here as well?

File: planner/core/common_plans.go [hunk 2]
```diff
@@ -834,8 +834,8 @@ func (h *AnalyzeTableID) Equals(t *AnalyzeTableID) bool {
 	return h.TableID == t.TableID && h.PartitionID == t.PartitionID
 }
 
-// analyzeInfo is used to store the database name, table name and partition name of analyze task.
-type analyzeInfo struct {
+// AnalyzeInfo is used to store the database name, table name and partition name of analyze task.
+type AnalyzeInfo struct {
```
Comment (qw4990) [inline planner/core/common_plans.go:838]: It seems that these lines will conflict with https://github.com/pingcap/tidb/pull/24999/files#diff-39be1be8697f0f187228689d867a84cb046ab2ff8ac643ac807244cdb7e2b842L837.

File: statistics/histogram.go (hunk 1 above)
Comment (winoros) [reply statistics/histogram.go:1288]: Oh it's debug will be removed.

File: planner/core/common_plans.go (hunk 2 above)
Comment (winoros) [reply planner/core/common_plans.go:838]: This is the same change actually(

File: planner/core/stats.go [hunk 3]
```diff
@@ -202,6 +202,7 @@ func (ds *DataSource) getGroupNDVs(colGroups [][]*expression.Column) []property.
 					Cols: idxCols,
 					NDV:  float64(idx.NDV),
 				}
+				logutil.BgLogger().Warn("get group ndv", zap.Int64s("cols", idxCols), zap.Float64("ndv", ndv.NDV))
```
Comment (qw4990) [inline planner/core/stats.go:205]: It seems a debug record, should we remove it?

Comment (winoros) [reply planner/core/stats.go:205]: Yes, will be removed when merge master or address other comments.

File: statistics/histogram.go [hunk 4]
```diff
@@ -298,10 +298,6 @@ const (
 	// Index stats: CM SKetch is not used. TopN and Histograms are built in TiKV using full data. NDV is also collected for each bucket in histogram.
 	//    Then values covered by TopN is removed from Histogram. TopN + Histogram represent all data.
 	Version2 = 2
-	// Version3 is used for testing now. Once it finished, we will fallback the Version3 to Version2.
-	// The difference between Version2 and Version3 is that we construct the index's statistics based on sampling also.
-	// The data structure between them are then same.
-	Version3 = 3
```
Comment (time-and-fate) [inline statistics/histogram.go:304]: Let's move some of these into Version2.

File: planner/core/testdata/analyze_suite_out.json [hunk 5]
```diff
@@ -435,14 +435,14 @@
   {
     "Name": "TestAnalyze",
     "Cases": [
-      "Analyze{Index(a),Table(a, b)}",
+      "Analyze{Table(_tidb_rowid, a, b, _tidb_rowid)}",
```
Comment (winoros) [inline planner/core/testdata/analyze_suite_out.json:438]: Currently, it's pruned by the execution phase.

**Review by qw4990 — APPROVED** [review]

File: statistics/handle/handle_test.go [hunk 6]
```diff
@@ -22,6 +22,8 @@ import (
 	"time"
 	"unsafe"
 
+	"github.com/pingcap/tidb/util/ranger"
```
Comment (time-and-fate) [inline statistics/handle/handle_test.go:25]: Need to be moved to order them alphabetically.

File: executor/analyze_test.go [hunk 7]
```diff
@@ -57,6 +57,7 @@ func (s *testSuite1) TestAnalyzePartition(c *C) {
 	testkit.WithPruneMode(tk, variable.Static, func() {
 		tk.MustExec("use test")
 		tk.MustExec("drop table if exists t")
+		tk.MustExec("set @@tidb_analyze_version=1")
```
[the remark this answers is not in the document]
Comment (winoros) [reply executor/analyze_test.go:60]: Actually nothing :joy:

**Review by time-and-fate — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]
lgtm

**Pull request merged**